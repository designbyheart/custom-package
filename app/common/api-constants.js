// @flow
// TODO:KS Consider moving it to api folder
export const TOKEN_EXPIRED_CODE = 'CRS-121'
export const SERVER_ERROR_CODE = 'OCS-000'
export const SERVER_API_CALL_ERROR = 'OCS-001'
export const PENDING_CONNECTION_REQUEST_CODE = 'CRS-103'
export const PUSH_NOTIFICATION_SENT_CODE = 'push-notification-sent'

export const QR_CODE_SIGNATURE = 's'
export const QR_CODE_CHALLENGE = 'c'
export const QR_CODE_REMOTE_CONNECTION_LOGO = 'lg'
export const QR_CODE_TARGET_DID = 'tDID'
export const QR_CODE_REMOTE_HOSTING_DID = 'rhDID'
export const QR_CODE_REMOTE_PAIR_WISE_DID = 'rpDID'
export const QR_CODE_TARGET_NAME = 'tn'
export const QR_CODE_SENDER_NAME = 'sn'
export const QR_CODE_UID = 'uid'

export const PUSH_NOTIFICATION_TYPE = {
  AUTH: 'auth-req',
  CLAIM_OFFER: 'claim-offer',
}
