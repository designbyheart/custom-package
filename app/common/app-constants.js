import { Platform } from 'react-native'

export const apptentiveCredentials = Platform.select({
  ios: {
    apptentiveKey: 'IOS-EVERNYM-ee7e2325084e',
    apptentiveSignature: '8049eba0f656a5b7aeb9722b84e5ec54',
  },
  android: {
    apptentiveKey: '<YOUR_ANDROID_APPTENTIVE_KEY>',
    apptentiveSignature: '<YOUR_ANDROID_APPTENTIVE_SIGNATURE>',
  },
})
