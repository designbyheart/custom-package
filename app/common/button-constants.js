export const ALLOW = 'Allow'
export const DENY = 'Deny'
