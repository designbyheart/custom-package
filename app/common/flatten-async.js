// @flow

// TODO:KS
// Explain the need for this function

export const flattenAsync = (
  fn: (...variableArgs: any[]) => Promise<any>
) => async (
  ...args: Array<any>
): Promise<[null | typeof Error, null | any]> => {
  try {
    const response = await fn(...args)
    return [null, response]
  } catch (e) {
    return [e, null]
  }
}
