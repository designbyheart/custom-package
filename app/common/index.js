// @flow
export * from './api-constants'
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
