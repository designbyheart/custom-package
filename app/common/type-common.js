// @flow

export type CustomError = {
  code: string,
  message: string,
}

export const INITIAL_TEST_ACTION: 'INITIAL_TEST_ACTION' = 'INITIAL_TEST_ACTION'

export type InitialTestAction = {
  type: typeof INITIAL_TEST_ACTION,
}

export const initialTestAction = () => ({
  type: INITIAL_TEST_ACTION,
})

export type NavigationParams = {
  [string]: any,
}

export type ReactNavigation = {
  navigation: {
    navigate: (route: string, params?: any) => void,
    state?: {
      params: NavigationParams,
    },
  },
}

export type GenericObject = {
  [string]: any,
}

export type GenericStringObject = {
  [string]: string,
}

export type MessageAnnotation = {
  name: string,
  version: string,
}

export type TopicAnnotation = {
  tid: number,
  mid: number,
}
