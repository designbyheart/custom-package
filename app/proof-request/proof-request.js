// @flow
import React, { PureComponent } from 'react'
import {
  StyleSheet,
  Platform,
  Image,
  FlatList,
  View,
  Alert,
} from 'react-native'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import {
  Container,
  CustomView,
  CustomButton,
  CustomText,
  Avatar,
  Icon,
  ImageColorPicker,
  ConnectionTheme,
  ClaimProofHeader,
  Separator,
  FooterActions,
} from '../components'
import { homeRoute } from '../common/'
import {
  OFFSET_2X,
  OFFSET_1X,
  color,
  OFFSET_3X,
  OFFSET_4X,
  OFFSET_5X,
  isiPhone5,
} from '../common/styles'
import type { Attribute } from '../push-notification/type-push-notification'
import type { ProofRequestAttributeListProp } from './type-proof-request'
import type { Store } from '../store/type-store'
import type {
  ProofRequestProps,
  AdditionalProofDataPayload,
} from './type-proof-request'
import ProofModal from './proof-modal'
import {
  rejectProofRequest,
  acceptProofRequest,
  ignoreProofRequest,
  proofRequestShown,
  getProof,
} from '../store'
import { getConnectionLogoUrl } from '../store/store-selector'
import { ERROR_CODE_MISSING_ATTRIBUTE } from '../proof/type-proof'

class ProofRequestAttributeList extends PureComponent<
  ProofRequestAttributeListProp,
  void
> {
  renderItem = ({ item, index }) => {
    const logoUrl = item.data
      ? item.claimUuid &&
        this.props.claimMap &&
        this.props.claimMap[item.claimUuid] &&
        this.props.claimMap[item.claimUuid].logoUrl
        ? { uri: this.props.claimMap[item.claimUuid].logoUrl }
        : require('../images/cb_evernym.png')
      : null

    return (
      <Container
        fifth
        style={[styles.attributeItem]}
        testID={`proof-request-attribute-item-${index}`}
        row
      >
        <Container fifth verticalSpace style={[styles.label]}>
          <CustomView fifth style={[styles.attributeListLabel]}>
            <CustomText
              h7
              uppercase
              semiBold
              bg="tertiary"
              transparentBg
              style={[styles.attributeListLabelText]}
            >
              {item.label}
            </CustomText>
          </CustomView>
          <CustomView fifth style={[styles.attributeListValue]}>
            <CustomText h6 demiBold bg="tertiary" transparentBg>
              {item.data}
            </CustomText>
          </CustomView>
        </Container>
        <Icon
          center
          medium
          round
          resizeMode="cover"
          src={logoUrl}
          testID={`proof-requester-logo-${index}`}
        />
      </Container>
    )
  }

  render() {
    const attributes: Array<Attribute> = this.props.list
    return (
      <Container fifth style={[styles.proofRequestData]}>
        <FlatList
          data={attributes}
          ItemSeparatorComponent={Separator}
          ListFooterComponent={Separator}
          renderItem={this.renderItem}
        />
      </Container>
    )
  }
}

export class ProofRequest extends PureComponent<ProofRequestProps, void> {
  close = () => {
    this.props.navigation.goBack()
  }

  onIgnore = () => {
    this.props.ignoreProofRequest(this.props.uid)
    this.close()
  }

  onReject = () => {
    this.props.rejectProofRequest(this.props.uid)
    this.close()
  }

  onSend = () => {
    this.props.acceptProofRequest(this.props.uid)
  }

  componentWillReceiveProps(nextProps: ProofRequestProps) {
    if (nextProps.proofGenerationError !== this.props.proofGenerationError) {
      if (
        nextProps.proofGenerationError &&
        nextProps.proofGenerationError.code === ERROR_CODE_MISSING_ATTRIBUTE
      ) {
        Alert.alert(
          'Missing information',
          nextProps.proofGenerationError.message
        )
      }
    }
  }

  componentDidMount() {
    this.props.proofRequestShown(this.props.uid)
    this.props.getProof(this.props.uid)
  }

  render() {
    const {
      data,
      name,
      uid,
      isValid,
      proofStatus,
      remotePairwiseDID,
      logoUrl,
      claimMap,
    } = this.props
    const testID = 'proof-request'
    const { name: title } = data
    const logoUri = logoUrl
      ? { uri: logoUrl }
      : require('../images/cb_evernym.png')

    return (
      <Container fifth>
        {isValid && (
          <ClaimProofHeader
            message={`${name} would like you to prove:`}
            title={title}
            onClose={this.onIgnore}
            logoUrl={logoUrl}
            testID={testID}
          >
            <Icon
              absolute="TopRight"
              src={require('../images/close.png')}
              small
              testID={`${testID}-icon-close`}
              onPress={this.close}
              style={[styles.headerCloseIconContainer]}
            />
            <CustomView fifth hCenter style={[styles.headerStripLogoContainer]}>
              <ConnectionTheme logoUrl={logoUrl} style={[styles.strip]} />
              <CustomView
                row
                vCenter
                spaceBetween
                style={[styles.avatarsContainer]}
                testID={`${testID}-text-avatars-container`}
              >
                <Icon
                  center
                  halo
                  extraLarge
                  resizeMode="cover"
                  src={require('../images/UserAvatar.png')}
                  style={[styles.verifierLogo]}
                  iconStyle={[styles.verifierLogoIcon]}
                  haloStyle={styles.logoHaloStyle}
                  testID={`${testID}-verifier-logo`}
                />
                <Image
                  style={[styles.checkMark]}
                  source={require('../images/icon_rightArrow.png')}
                  testID={`${testID}-check-mark`}
                />
                <Icon
                  center
                  halo
                  extraLarge
                  resizeMode="cover"
                  src={logoUri}
                  haloStyle={styles.logoHaloStyle}
                  style={[styles.verifierLogo]}
                  iconStyle={[styles.verifierLogoIcon]}
                  testID={`${testID}-verifier-logo`}
                />
              </CustomView>
            </CustomView>
          </ClaimProofHeader>
        )}
        {isValid ? (
          <ProofRequestAttributeList
            list={data.requestedAttributes}
            claimMap={claimMap}
          />
        ) : (
          <Container fifth center>
            <CustomText h5 bg="fifth">
              Invalid proof request. Please ignore.
            </CustomText>
          </Container>
        )}
        <FooterActions
          logoUrl={logoUrl}
          onAccept={this.onSend}
          onDecline={this.onReject}
          denyTitle="Ignore"
          acceptTitle="Send"
          testID={testID}
        />
        {isValid && (
          <ProofModal
            proofStatus={proofStatus}
            title={title}
            name={name}
            onContinue={this.close}
            logoUrl={logoUrl}
          />
        )}
      </Container>
    )
  }
}

// we need to navigate to this screen only by passing "proofRequestId"  in the props like below.
// this.props.navigation.navigate(proofRequestRoute,{'proofRequestId':'CRM2M28')
const mapStateToProps = (state: Store, props) => {
  const { proofRequest } = state
  const { uid } = props.navigation.state.params
  const proofRequestData = proofRequest[uid] || {}
  const {
    data,
    requester = {},
    proofStatus,
    status,
    remotePairwiseDID,
  } = proofRequestData
  const { name } = requester
  const isValid = proofRequestData && data && data.requestedAttributes
  const proofGenerationError = state.proof[uid] ? state.proof[uid].error : null

  return {
    isValid,
    data,
    logoUrl: getConnectionLogoUrl(state, remotePairwiseDID),
    name,
    uid,
    proofStatus,
    proofGenerationError,
    claimMap: state.claim.claimMap,
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      proofRequestShown,
      acceptProofRequest,
      ignoreProofRequest,
      rejectProofRequest,
      getProof,
    },
    dispatch
  )

export default connect(mapStateToProps, mapDispatchToProps)(ProofRequest)

const styles = StyleSheet.create({
  headerStripLogoContainer: {
    height: 90,
    marginHorizontal: isiPhone5 ? OFFSET_5X / 2 : OFFSET_4X,
  },
  strip: {
    position: 'absolute',
    left: isiPhone5 ? OFFSET_3X : OFFSET_5X,
    right: isiPhone5 ? OFFSET_3X : OFFSET_5X,
    height: 30,
    backgroundColor: color.actions.secondary,
    zIndex: -1,
  },
  verifierLogo: {
    height: 70,
    zIndex: 1,
  },
  verifierLogoIcon: {
    borderRadius: 35,
    height: 70,
    width: 70,
  },
  logoHaloStyle: {
    width: 87,
    height: 87,
  },
  proofRequestData: {
    zIndex: 3,
  },
  attributeListLabel: {
    marginVertical: 2,
  },
  attributeListLabelText: {
    paddingVertical: 2,
  },
  attributeListValue: {
    marginBottom: OFFSET_1X,
  },
  headerCloseIconContainer: {
    position: 'absolute',
    top: OFFSET_3X,
    right: OFFSET_3X / 2,
    zIndex: 3,
  },
  avatarsContainer: {
    marginVertical: OFFSET_3X,
    marginHorizontal: OFFSET_1X,
  },
  checkMark: {
    width: 20,
    height: 20,
  },
  attributeItem: {
    paddingHorizontal: OFFSET_1X,
  },
  label: {
    marginLeft: OFFSET_1X / 2,
  },
})
