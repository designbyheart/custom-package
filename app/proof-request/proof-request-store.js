// @flow
import { put, takeLatest, call, select, all } from 'redux-saga/effects'
import type { CustomError } from '../common/type-common'
import type {
  ProofRequestStore,
  ProofRequestAction,
  ProofRequestShownAction,
  SendProofSuccessAction,
  SendProofFailAction,
  SendProofAction,
  ProofRequestIgnoredAction,
  ProofRequestRejectedAction,
  ProofRequestAcceptedAction,
  ProofRequestPayload,
  AdditionalProofDataPayload,
  MissingAttribute,
  SelfAttestedAttributes,
  MissingAttributes,
} from './type-proof-request'

import {
  getUserPairwiseDid,
  getAgencyUrl,
  getProof,
  getProofRequestPairwiseDid,
  getUserOneTimeInfo,
  getAgencyVerificationKey,
  getRemotePairwiseDidAndName,
  getPoolConfig,
} from '../store/store-selector'
import {
  PROOF_REQUEST_RECEIVED,
  PROOF_REQUEST_STATUS,
  PROOF_STATUS,
  PROOF_REQUEST_SHOWN,
  PROOF_REQUEST_IGNORED,
  PROOF_REQUEST_ACCEPTED,
  PROOF_REQUEST_REJECTED,
  PROOF_REQUEST_AUTO_FILL,
  SEND_PROOF,
  SEND_PROOF_SUCCESS,
  SEND_PROOF_FAIL,
  MISSING_ATTRIBUTES_FOUND,
} from './type-proof-request'
import type {
  NotificationPayloadInfo,
  Attribute,
} from '../push-notification/type-push-notification'
import { sendMessage } from '../bridge/react-native-cxs/RNCxs'
import type { Connection } from '../store/type-connection-store'
import type { UserOneTimeInfo } from '../store/user/type-user-store'
import { MESSAGE_TYPE } from '../api/api-constants'
import { RESET } from '../common/type-common'

const proofRequestInitialState = {}

export const ignoreProofRequest = (uid: string): ProofRequestIgnoredAction => ({
  type: PROOF_REQUEST_IGNORED,
  uid,
})
export const rejectProofRequest = (
  uid: string
): ProofRequestRejectedAction => ({
  type: PROOF_REQUEST_REJECTED,
  uid,
})
export const acceptProofRequest = (
  uid: string
): ProofRequestAcceptedAction => ({
  type: PROOF_REQUEST_ACCEPTED,
  uid,
})
export const proofRequestShown = (uid: string): ProofRequestShownAction => ({
  type: PROOF_REQUEST_SHOWN,
  uid,
})
export const sendProof = (uid: string): SendProofAction => ({
  type: SEND_PROOF,
  uid,
})
export const sendProofSuccess = (uid: string): SendProofSuccessAction => ({
  type: SEND_PROOF_SUCCESS,
  uid,
})

export const sendProofFail = (
  uid: string,
  error: CustomError
): SendProofFailAction => ({
  type: SEND_PROOF_FAIL,
  error,
  uid,
})

export function convertMissingAttributeListToObject(
  missingAttributes: Array<MissingAttribute>
): MissingAttributes {
  return missingAttributes.reduce(
    (selfAttestedAttributes, missingAttribute: MissingAttribute) => ({
      ...selfAttestedAttributes,
      [missingAttribute.name.toLocaleLowerCase()]: {
        name: missingAttribute.name,
        data: '',
        key: missingAttribute.key,
      },
    }),
    {}
  )
}

export const missingAttributesFound = (
  missingAttributeList: MissingAttribute[],
  uid: string
) => ({
  type: MISSING_ATTRIBUTES_FOUND,
  missingAttributes: convertMissingAttributeListToObject(missingAttributeList),
  uid,
})

export function* watchProofRequestAccepted(): Generator<*, *, *> {
  yield takeLatest(PROOF_REQUEST_ACCEPTED, proofAccepted)
}

export function* proofAccepted(
  action: ProofRequestAcceptedAction
): Generator<*, *, *> {
  const remoteDid: string = yield select(getProofRequestPairwiseDid, action.uid)
  const userPairwiseDid: string | null = yield select(
    getUserPairwiseDid,
    remoteDid
  )

  if (userPairwiseDid) {
    // set status that we are generating and sending proof
    yield put(sendProof(action.uid))
    try {
      const agencyUrl: string = yield select(getAgencyUrl)
      const poolConfig: string = yield select(getPoolConfig)
      const messageId: string = action.uid
      const payload = yield select(getProof, messageId)
      const proof = {
        ...payload,
        remoteDid,
        userPairwiseDid,
      }
      const agencyVerificationKey: string = yield select(
        getAgencyVerificationKey
      )
      const connection: Connection = yield select(
        getRemotePairwiseDidAndName,
        userPairwiseDid
      )
      const userOneTimeInfo: UserOneTimeInfo = yield select(getUserOneTimeInfo)

      const url = `${agencyUrl}/agency/msg`
      try {
        const sendProofStatus = yield call(sendMessage, {
          url,
          messageType: MESSAGE_TYPE.PROOF,
          messageReplyId: messageId,
          message: JSON.stringify(proof),
          myPairwiseDid: connection.myPairwiseDid,
          myPairwiseVerKey: connection.myPairwiseVerKey,
          myPairwiseAgentDid: connection.myPairwiseAgentDid,
          myPairwiseAgentVerKey: connection.myPairwiseAgentVerKey,
          myOneTimeAgentDid: userOneTimeInfo.myOneTimeAgentDid,
          myOneTimeAgentVerKey: userOneTimeInfo.myOneTimeAgentVerificationKey,
          myOneTimeDid: userOneTimeInfo.myOneTimeDid,
          myOneTimeVerKey: userOneTimeInfo.myOneTimeVerificationKey,
          myAgencyVerKey: agencyVerificationKey,
          myPairwisePeerVerKey: connection.myPairwisePeerVerKey,
          poolConfig,
        })
        yield put(sendProofSuccess(action.uid))
      } catch (e) {
        yield put(sendProofFail(action.uid, e))
      }
    } catch (e) {
      yield put(sendProofFail(action.uid, e))
    }
  } else {
    yield put(
      sendProofFail(action.uid, {
        code: 'OCS-002',
        message: 'No pairwise connection found',
      })
    )
  }
}

export const proofRequestAutoFill = (
  uid: string,
  requestedAttributes: Array<Attribute>
) => ({
  type: PROOF_REQUEST_AUTO_FILL,
  uid,
  requestedAttributes,
})

export const proofRequestReceived = (
  payload: AdditionalProofDataPayload,
  payloadInfo: NotificationPayloadInfo
) => ({
  type: PROOF_REQUEST_RECEIVED,
  payload,
  payloadInfo,
})

export default function proofRequestReducer(
  state: ProofRequestStore = proofRequestInitialState,
  action: ProofRequestAction
) {
  switch (action.type) {
    case PROOF_REQUEST_RECEIVED:
      return {
        ...state,
        [action.payloadInfo.uid]: {
          ...action.payload,
          ...action.payloadInfo,
          status: PROOF_REQUEST_STATUS.RECEIVED,
          proofStatus: PROOF_STATUS.NONE,
        },
      }

    case PROOF_REQUEST_SHOWN:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          status: PROOF_REQUEST_STATUS.SHOWN,
        },
      }

    case PROOF_REQUEST_ACCEPTED:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          status: PROOF_REQUEST_STATUS.ACCEPTED,
        },
      }

    case PROOF_REQUEST_IGNORED:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          status: PROOF_REQUEST_STATUS.IGNORED,
        },
      }

    case PROOF_REQUEST_REJECTED:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          status: PROOF_REQUEST_STATUS.REJECTED,
        },
      }

    case PROOF_REQUEST_AUTO_FILL:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          data: {
            ...state[action.uid].data,
            requestedAttributes: [...action.requestedAttributes],
          },
        },
      }

    case SEND_PROOF:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          proofStatus: PROOF_STATUS.SENDING_PROOF,
        },
      }

    case SEND_PROOF_SUCCESS:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          proofStatus: PROOF_STATUS.SEND_PROOF_SUCCESS,
        },
      }

    case SEND_PROOF_FAIL:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          proofStatus: PROOF_STATUS.SEND_PROOF_FAIL,
        },
      }

    case MISSING_ATTRIBUTES_FOUND:
      return {
        ...state,
        [action.uid]: {
          ...state[action.uid],
          missingAttributes: action.missingAttributes,
        },
      }
    case RESET:
      return proofRequestInitialState
    default:
      return state
  }
}
