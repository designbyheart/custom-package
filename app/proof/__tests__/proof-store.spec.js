// @flow
import renderer from 'react-test-renderer'
import { put, call } from 'redux-saga/effects'
import { initialTestAction } from '../../common/type-common'
import proofReducer, { proofSuccess } from '../proof-store'
import { acceptProofRequest } from '../../proof-request/proof-request-store'
import {
  prepareProof,
  generateProof,
} from '../../bridge/react-native-cxs/RNCxs'

describe('Proof Store', () => {
  const proofRequest = {
    nonce: '123432421212',
    name: 'proof_req_1',
    version: '0.1',
    requested_attrs: {
      attr1_uuid: {
        schema_seq_no: 103,
        issuer_did: 'V4SGRU86Z58d6TV7PBUe6f',
        name: 'name',
      },
    },
    requested_predicates: {},
  }
  const remoteDid = 'LXemRr9feto7vMRcaoby7f'
  const uid = '1234'

  let initialState
  let afterProofSuccess
  beforeEach(() => {
    initialState = proofReducer(undefined, initialTestAction())
    afterProofSuccess = proofReducer(
      initialState,
      proofSuccess(proofRequest, uid)
    )
  })

  it('should match snapshot for proof received action', () => {
    expect(afterProofSuccess).toMatchSnapshot()
  })

  xit('show fail if attributes are missing from claims', () => {
    // TODO:KS Add implementation once connection history test related
    // changes are available, want to avoid bit of rework
  })
})
