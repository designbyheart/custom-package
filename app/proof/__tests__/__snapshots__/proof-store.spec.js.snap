// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proof Store fn:convertIndyRequestedProofToAttributes 1`] = `
Array [
  Object {
    "claimUuid": "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
    "data": "Alex",
    "label": "Address 1",
  },
  Object {
    "claimUuid": "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
    "data": "male",
    "label": "Address 2",
  },
]
`;

exports[`Proof Store fn:convertIndyRequestedProofToAttributes 2`] = `
Array [
  Object {
    "claimUuid": "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
    "data": "Alex",
    "label": "Address 1",
  },
  Object {
    "claimUuid": "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
    "data": "male",
    "label": "Address 2",
  },
  Object {
    "data": "attr 3 value",
    "label": "Address 3",
  },
  Object {
    "data": "attr 4 value",
    "label": "Address 4",
  },
  Object {
    "data": "attr 5 value",
    "label": "Address 5",
  },
  Object {
    "data": "attr 6 value",
    "label": "Address 6",
  },
  Object {
    "data": "attr 7 value",
    "label": "Address 7",
  },
  Object {
    "data": "attr 8 value",
    "label": "Address 8",
  },
  Object {
    "data": "attr 9 value",
    "label": "Address 9",
  },
  Object {
    "data": "attr 10 value",
    "label": "Address 10",
  },
]
`;

exports[`Proof Store match snapshot while converting SelfAttestedAttributes to Indy format 1`] = `
Object {
  "attr2_uuid": "male",
}
`;

exports[`Proof Store match snapshot while converting SelfAttestedAttributes to Indy format 2`] = `
Object {
  "attr2_uuid": "male",
  "attr3_uuid": "190",
}
`;

exports[`Proof Store should convert prepared proof to indy proof format 1`] = `
Array [
  Object {
    "attr1_uuid": Array [
      "claim::ea03d8ca-eeb4-4944-b7d6-5abcf4503d73",
      true,
    ],
    "attr2_uuid": Array [
      "claim::6a0f42b4-1210-4bdb-ad53-10ed765276b5",
      true,
    ],
  },
  Array [],
]
`;

exports[`Proof Store should find missing attributes if any missing 1`] = `
Array [
  Object {
    "attr1_uuid": Array [
      "claim::ea03d8ca-eeb4-4944-b7d6-5abcf4503d73",
      true,
    ],
    "attr3_uuid": Array [
      "claim::6a0f42b4-1210-4bdb-ad53-10ed765276b5",
      true,
    ],
  },
  Array [
    Object {
      "key": "attr2_uuid",
      "name": "address2",
    },
  ],
]
`;

exports[`Proof Store should match snapshot for proof received action 1`] = `
Object {
  "pid123": Object {
    "aggregated_proof": Object {
      "c_hash": "31",
      "c_list": Array [
        Array [
          36,
        ],
      ],
    },
    "proofs": Object {
      "claim::f760213b-e719-47fb-9669-b72c5c410e8c": Object {
        "issuer_did": "issuerDid",
        "proof": Object {
          "non_revoc_proof": null,
          "primary_proof": Object {
            "eq_proof": Object {
              "a_prime": "13",
              "e": "51",
              "m": Object {},
              "m1": "67",
              "m2": "60",
              "revealed_attrs": Object {
                "name": "11",
                "sex": "59",
              },
              "v": "13",
            },
            "ge_proofs": Array [],
          },
        },
        "schema_seq_no": 36,
      },
    },
    "requested_proof": Object {
      "predicates": Object {},
      "revealed_attrs": Object {
        "attr1_uuid": Array [
          "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
          "Alex",
          "11",
        ],
        "attr2_uuid": Array [
          "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
          "male",
          "59",
        ],
      },
      "self_attested_attrs": Object {},
      "unrevealed_attrs": Object {},
    },
  },
}
`;

exports[`Proof Store store update correctly if proof fails 1`] = `
Object {
  "pid123": Object {
    "error": Object {
      "code": "TEST-CODE",
      "message": "test error message",
    },
  },
}
`;
